spark_pipeline_framework/transformers/http_data_sender/v3/http_data_sender.py

```diff
@@ -242,9 +248,9 @@ def _transform(self, df: DataFrame) -> DataFrame:
             self.logger.info(f"Total Batches: {desired_partitions}")
 
             # ---- Now process all the results ----
-            rdd: RDD[Row] = df.repartition(
-                desired_partitions
-            ).rdd.mapPartitionsWithIndex(
+            if partition_data:
+                df = df.repartition(desired_partitions)
```

Comment:
If we're not doing repartitioning, then we are doing incorrect logging of desired partitions above. And we should avoid calculating the desired partitions in case of no repartitioning.